project_generation/project_generation.go

```diff
@@ -288,5 +290,12 @@ func (a application) generateFileFromTemplate(fileToGen fileGen) (err error) {
 	if err != nil {
 		return err
 	}
+
+	if fileToGen.executable {
```

Comment:
You could probably extract this into a separate reusable function.

Reply:
If I do then it won't make the code any simpler as it will still ned to evaluate returned errors. Effectively it would just be a thin wrapper around `os.Chmod`. I intend to keep it as is.